pkg/apis/cloudcredential/v1/aws_types.go

```diff
@@ -49,3 +51,71 @@ type AWSProviderStatus struct {
 	// Policy is the name of the policy attached to the user in AWS.
 	Policy string `json:"policy"`
 }
+
+// AWSCondition - map of condition types, with associated key - value mapping
+type AWSCondition map[AWSConditionType]AWSConditionKeyValue
+
+// AWSConditionKeyValue - mapping of values for the chousen type
+type AWSConditionKeyValue map[string]interface{}
+
+// AWSConditionType provides condition operators to match the condition key and value in the policy against values in the request context
+type AWSConditionType string
+
+const (
+	// String elements that restrict access based on comparing a key to a string value.
+	String AWSConditionType = "String"
+	// Numeric elements that restrict access based on comparing a key to an integer or decimal value.
+	Numeric AWSConditionType = "Numeric"
+	// Date elements that restrict access based on comparing a key to a date/time value.
+	Date AWSConditionType = "Date"
+	// Bool elements that restrict access based on comparing a key to "true" or "false."
+	Bool AWSConditionType = "Bool"
+	// Arn elements that restrict access based on comparing a key to an ARN.
+	Arn AWSConditionType = "Arn"
+	// Null check if a condition key is present at the time of authorization.
+	Null AWSConditionType = "Null"
+	// Binary elements that test key values that are in binary format.
+	Binary AWSConditionType = "Binary"
+	// IPAddress restrict access based on comparing a key to an IPv4 or IPv6 address or range of IP addresses.
+	IPAddress AWSConditionType = "IpAddress"
+)
+
+// Not prepends Not prefix to the condition
+func (c AWSConditionType) Not() AWSConditionType {
+	return "Not" + c
+}
+
+// IgnoreCase appends IgnoreCase prefix to the condition
+func (c AWSConditionType) IgnoreCase() AWSConditionType {
+	return c + "IgnoreCase"
+}
+
+// Equals appends Equals prefix to the condition
+func (c AWSConditionType) Equals() AWSConditionType {
+	return c + "Equals"
+}
+
+// NotEquals appends NotEquals prefix to the condition
+func (c AWSConditionType) NotEquals() AWSConditionType {
+	return c + "NotEquals"
```

Comment:
Would a condition type of `StringNotEquals` unmarshal from JSON correctly?

Reply:
It totally will, as it will undergo only a raw encoding to internal golang structure, where the types are still raw strings.